--- faye-browser.js.orig	2016-02-15 15:43:20.000000000 -0800
+++ faye-browser.js	2016-04-28 10:45:51.143056190 -0700
@@ -9,7 +9,7 @@
   JSONP_CALLBACK:   'jsonpcallback',
   CONNECTION_TYPES: ['long-polling', 'cross-origin-long-polling', 'callback-polling', 'websocket', 'eventsource', 'in-process'],
 
-  MANDATORY_CONNECTION_TYPES: ['long-polling', 'callback-polling', 'in-process'],
+  MANDATORY_CONNECTION_TYPES: ['websocket', 'long-polling', 'callback-polling', 'in-process'],
 
   ENV: (typeof window !== 'undefined') ? window : global,
 
@@ -1158,6 +1158,13 @@
     this.info('Initiating handshake with ?', Faye.URI.stringify(this.endpoint));
     this._selectTransport(Faye.MANDATORY_CONNECTION_TYPES);
 
+    if (!this._transport) {
+      this.info('Transport not yet ready');
+      Faye.ENV.setTimeout(function() { self.handshake(callback, context) }, this._advice.interval);
+      this._state = this.UNCONNECTED;
+      return;
+    }
+
     this._send({
       channel:                  Faye.Channel.HANDSHAKE,
       version:                  Faye.BAYEUX_VERSION,
